detectable/src/main/java/com/blackduck/integration/detectable/detectables/uv/transform/UVLockParser.java

```diff
@@ -146,6 +147,13 @@ private void generateGraph(UVDetectorOptions uvDetectorOptions) {
 
     // parse all dependencies which we had stored in the Map for transitive dependency for each dependency while parsing it the first time
     private void loopOverDependencies(String dependency, Dependency parentDependency, UVDetectorOptions uvDetectorOptions) {
+
+        if(visitedDependencies.contains(dependency)) {
```

Comment:
Is it possible to encounter, since it is a String, the same dependency in trees that don't contain a cycle? Like direct dependency A bringing in transitive dependency C and B also bringing in dependency C? Not fully sure what this code does and if that concern is valid or accounted for.